### PR TITLE
V3/controllers harvest auth

### PR DESCRIPTION
**Pull request by leefrank9527**
1. Fixed validation error presentation page: replaced Spring Tags with customized error parameters.
2. Fixed controllers:  a) TabbedController: extracted a method to do initBind, binding and validation.
  b) Controllers of Harvest Authorisations, all controllers in package: org.webcurator.ui.site.controller.
  c) Fixed issues of target related controllers to match changed TabbedController.
  d) Commented useless mapping lines in ServletConfig.

### CODE REVIEW

**Review by hannakoppelaar — APPROVED** [review]
I think this is okay. @obrienben The related PR for core is yet to be reviewed by you. I guess they should be merged at the same time, since the changes in webapp depend on those in core. If you're happy with the PR for core, I'm fine with you merging both PRs to their respective master branches.

File: src/main/java/org/webcurator/ui/site/controller/SiteController.java [hunk 1]
```diff
@@ -53,46 +55,55 @@
...
 
 /**
  * The Controller for managing the Harvest Autorisation tabs.
+ *
  * @author bbeaumont
  */
 @Controller
 @Scope(BeanDefinition.SCOPE_SINGLETON)
 @Lazy(false)
+@RequestMapping("/curator/site/site.html")
```
Comment (obrienben) [inline src/main/java/org/webcurator/ui/site/controller/SiteController.java:70]: Can you compare this line with some of the other controllers we have converted, I think we were dropping the ".html" from the end of the request mapping path.

File: src/main/java/org/webcurator/ui/site/controller/SitePermissionController.java [hunk 2]
```diff
@@ -53,389 +56,387 @@
...
+        }
+    }
+
+
+    /**
+     * Handle the request by sending it to the appropriate sub-handler.
+     *
+     * @param req           The HttpServletRequest object.
+     * @param resp          The HttpServletResponse object.
+     * @param comm          The Spring command object.
```
Comment (obrienben) [inline src/main/java/org/webcurator/ui/site/controller/SitePermissionController.java]: Doesn't match the new name of the parameter - should now be 'command'.
Probably not worthy of a commit.

File: src/main/java/org/webcurator/ui/site/controller/SiteSearchController.java [hunk 3]
```diff
@@ -45,131 +49,134 @@
 
 /**
  * The controller for managing searching for harvest authorisations.
+ *
  * @author bbeaumont
  */
 @Controller
 @Scope(BeanDefinition.SCOPE_SINGLETON)
 @Lazy(false)
+@RequestMapping("/curator/site/search.html")
```
Comment (obrienben) [inline src/main/java/org/webcurator/ui/site/controller/SiteSearchController.java:58]: See previous comment about dropping the ".html" from request mapping path

File: src/main/java/org/webcurator/ui/target/controller/TabbedTargetInstanceController.java [hunk 4]
```diff
@@ -53,6 +54,7 @@
 @Controller
 @Scope(BeanDefinition.SCOPE_SINGLETON)
 @Lazy(false)
+@RequestMapping("/curator/target/target-instance.html")
```
Comment (obrienben) [inline src/main/java/org/webcurator/ui/target/controller/TabbedTargetInstanceController.java:57]: Same comment about the ".html"

File: src/main/java/org/webcurator/ui/target/controller/TargetController.java [hunk 5]
```diff
@@ -5,10 +5,20 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 @Controller
 @Scope(BeanDefinition.SCOPE_SINGLETON)
 @Lazy(false)
 @RequestMapping("/curator/target/target.html")
```
Comment (obrienben) [inline src/main/java/org/webcurator/ui/target/controller/TargetController.java:17]: Same comment about the ".html"

**Review by obrienben — CHANGES_REQUESTED** [review]
@leefrank9527 two buttons that I noticed didn't work. Did you want to fix and include them in this PR?

Transferring a permission
image

image

And generating letter/email
image
image

File: src/main/java/org/webcurator/ui/site/controller/SitePermissionController.java (hunk 2 above)
Comment (leefrank9527) [reply src/main/java/org/webcurator/ui/site/controller/SitePermissionController.java]: Accepted. Changed the param in comments to match the actual parameter in the function.

**Comment by obrienben** [comment]
@leefrank9527 anything else to include in this PR or is it done?

**Comment by frank-dia** [comment]
@obrienben It's done.

**Comment by frank-dia** [comment]
@obrienben It's done.

**Pull request merged**